### PR TITLE
SONARPY-625 Avoid throwing when highlighting symbols in nested f-string

### PR DESCRIPTION
**Pull request by pynicolas**

### CODE REVIEW

File: python-frontend/src/main/java/org/sonar/python/lexer/FStringChannel.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ */
+package org.sonar.python.lexer;
+
+import com.sonar.sslr.api.GenericTokenType;
+import com.sonar.sslr.api.Token;
+import com.sonar.sslr.impl.Lexer;
+import org.sonar.sslr.channel.Channel;
+import org.sonar.sslr.channel.CodeReader;
+
+public class FStringChannel extends Channel<Lexer> {
```
Comment (andrea-guarino-sonarsource) [inline python-frontend/src/main/java/org/sonar/python/lexer/FStringChannel.java]: Please add a comment explaining this Channel is useful to handle "literal_char" in f-string

File: python-frontend/src/main/java/org/sonar/python/tree/StringElementImpl.java [hunk 2]
```diff
@@ -53,7 +63,7 @@ public void accept(TreeVisitor visitor) {
 
   @Override
   public List<Tree> computeChildren() {
-    return Collections.singletonList(token);
+    return Stream.concat(Stream.of(token), formattedExpressions.stream()).collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (pynicolas) [reply python-frontend/src/main/java/org/sonar/python/tree/StringElementImpl.java]: Done in 0cc6ffb

File: python-frontend/src/main/java/org/sonar/python/api/PythonGrammar.java [hunk 3]
```diff
@@ -199,7 +201,15 @@ public static void grammar(LexerfulGrammarBuilder b) {
 
     b.rule(STAR_EXPR).is("*", EXPR);
     b.rule(EXPR).is(XOR_EXPR, b.zeroOrMore("|", XOR_EXPR));
-    b.rule(FORMATTED_EXPR).is(b.sequence(EXPR, b.optional(PythonPunctuator.ASSIGN)));
+
+    b.rule(F_STRING_CONTENT).is(b.zeroOrMore(b.firstOf(GenericTokenType.UNKNOWN_CHAR, FORMATTED_EXPR)));
```
[the remark this answers is not in the document]
Comment (pynicolas) [reply python-frontend/src/main/java/org/sonar/python/api/PythonGrammar.java]: Done in 0cc6ffb

File: python-frontend/src/main/java/org/sonar/python/lexer/FStringChannel.java (hunk 1 above)
Comment (pynicolas) [reply python-frontend/src/main/java/org/sonar/python/lexer/FStringChannel.java]: Done in 0cc6ffb

**Pull request merged**